### PR TITLE
Feat: initial contracts folder setup, registry structs

### PR DESCRIPTION
**Pull request by hstove**
- Closes https://github.com/stacks-network/sbtc/issues/112

## Description

This PR introduces the `contracts` folder, which is where sBTC Clarity contracts and contract unit tests will be located.

Within that folder is the start of the `sbtc-registry` contract, which is where data structures for deposits and withdrawals will live. This PR implements the data structures for withdrawals, along with a function and tests for writing new withdrawal requests.

Note that this does not implement the user-facing `initiate-withdrawal-request` contract function. Instead, this is only an "internal" function for storing the necessary data.

The unit tests are located in [`./contracts/tests/sbtc-registry.test.ts`](./contracts/tests/sbtc-registry.test.ts).

### PNPM workspace

This also introduces a [PNPM workspace](https://pnpm.io/workspaces), which allows for efficiently installing and managing multiple folders that include JS packages. It also allows easily running a command multiple workspace packages at once (such as `pnpm test -r` to run `test` in each package).

## Type of Change

- New feature

## Does this introduce a breaking change?
List the APIs or describe the functionality that this PR breaks.
Workarounds for or expected timeline for deprecation

## Are documentation updates required?

- Link to documentation updates: 
  - Top-level README for installing contract-related dependencies
  - Contracts folder README for setup and testing instructions related to contracts

## Testing information

- Tests can be performed by running `pnpm test` in the contracts folder.

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `pnpm test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @person1 or @person2

### CODE REVIEW

File: contracts/contracts/sbtc-registry.clar [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    (try! (validate-caller))
+    ;; #[allow(unchecked_data)]
+    (map-set withdrawal-requests id {
+      amount: amount,
+      max-fee: max-fee,
+      sender: sender,
+      recipient: recipient,
+      block-height: height,
+    })
+    (ok id)
```
[the remark this answers is not in the document]
Comment (hstove) [reply contracts/contracts/sbtc-registry.clar:61]: Yep you're right, I did mean to include events in this function. I'll update. Good catch!

**Comment by AshtonStephens** [comment]
A note on this template, @hstove I assumed this template was from the `stacks-core` repo for which most of the template didn't apply; do you think we should be filling out the PR template description the way it asks?

**Comment by hstove** [comment]
@AshtonStephens I'm not quite sure what you mean - that we should or shouldn't use this template? Or that this PR doesn't follow the template well enough?

**Comment by hstove** [comment]
Added `print` event and related tests in c669e3f

**Review by setzeus — APPROVED** [review]
LGTM! Kudos on the setup.

**Comment by AshtonStephens** [comment]
> @AshtonStephens I'm not quite sure what you mean - that we should or shouldn't use this template? Or that this PR doesn't follow the template well enough?

I'm just asking you if we want to, but I'm not suggesting either way.

**Comment by hstove** [comment]
@AshtonStephens I found the template to be pretty helpful - I didn't follow it to the tee, but most of it seemed pretty applicable. That's partly because, for example, this PR actually needs testing instructions, but in the future we'll know how to test (and they'll be in CI). There are some things to probably remove (ie "tag X and Y"), but otherwise I like it.

In general, I think templates are overall helpful, as long as they don't have to be 100% matched 100% of the time.

**Pull request merged**